src/app/modules/admin-feedback/admin-feedback.service.ts

```diff
@@ -0,0 +1,118 @@
+import { isEmpty } from 'lodash'
+import mongoose from 'mongoose'
+import { errAsync, okAsync, ResultAsync } from 'neverthrow'
+
+import { createLoggerWithLabel } from '../../config/logger'
+import getAdminFeedbackModel from '../../models/admin_feedback.server.model'
+import { DatabaseError } from '../core/core.errors'
+
+import { MissingAdminFeedbackError } from './admin-feedback.errors'
+
+const AdminFeedbackModel = getAdminFeedbackModel(mongoose)
+const logger = createLoggerWithLabel(module)
+
+/**
+ * Inserts given admin feedback to the database.
+ * @param userId the userId of the admin that provided the feedback
+ * @param rating the feedback rating to insert (0 for thumbs down, 1 for thumbs up)
+ * @param comment the feedback comment to insert if available
+ * @returns ok(IAdminFeedbackSchema) if successfully inserted
+ * @returns err(DatabaseError) on database error
+ */
+export const insertAdminFeedback = ({
+  userId,
+  rating,
+  comment,
+}: {
+  userId: string
+  rating: number
+  comment?: string
+}) => {
+  return ResultAsync.fromPromise(
+    AdminFeedbackModel.create({
+      userId,
+      rating,
+      comment,
+    }),
+    (error) => {
+      logger.error({
+        message: 'Database error when creating admin feedback document',
+        meta: {
+          action: 'insertAdminFeedback',
+          userId,
+        },
+        error,
+      })
+
+      return new DatabaseError('Admin feedback could not be created')
+    },
+  )
+}
+
+/**
+ * Updates admin feedback in the database.
+ * Will use previous value if comment or rating are not passed into method
```

Comment:
/nit technically it should be `will not update the previous value if comment or rating are not passed into method`?

Reply:
Thank you!! [8405543](https://github.com/opengovsg/FormSG/pull/6454/commits/84055434476aa416b0af672dc402825f3c6ae14d)